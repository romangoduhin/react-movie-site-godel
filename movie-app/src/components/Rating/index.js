import React from 'react';
import PropTypes from 'prop-types';
import style from './index.module.scss';

function Rating({ vote }) {
  return <div className={style.vote}>{vote}</div>;
}

Rating.defaultProps = {
  vote: 0,
};

Rating.propTypes = {
  vote: PropTypes.number,
};

export default Rating;
