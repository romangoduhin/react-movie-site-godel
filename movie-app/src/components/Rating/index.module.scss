@import "../../assets/styles/variables";
@import "../../assets/styles/mixins";


.vote {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 50px;
  width: 50px;

  color: $white;
  background-color: $orange;
  border-radius: 30px;
}