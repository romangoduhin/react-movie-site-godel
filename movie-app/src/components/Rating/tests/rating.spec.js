import React from 'react';
import Rating from '../index';

const props = {
  vote: 2,
};

describe('Rating component', () => {
  it('should render Rating component with props', () => {
    const component = shallow(<Rating {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should render Rating component without props', () => {
    const component = shallow(<Rating />);
    expect(component).toMatchSnapshot();
  });
});
