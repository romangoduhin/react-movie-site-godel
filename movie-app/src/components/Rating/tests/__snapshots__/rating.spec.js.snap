// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component should render Rating component with props 1`] = `
<div
  className="vote"
>
  2
</div>
`;

exports[`Rating component should render Rating component without props 1`] = `
<div
  className="vote"
>
  0
</div>
`;
